### PR TITLE
Mark branch for created PR.

### PR DESCRIPTION
**Pull request by grokys**
# Description

When we create a pull request using the GitHub pane, mark the branch in the `.git/config` to associate the branch with the pull request, meaning that the status bar shows the PR for the current branch.

# Testing

- Create a new branch
- Make some changes and commit
- Create a new Pull Request using the link in the PR list to create a new PR
- Previously, the new PR would not be highlighted in the PR list, and would not show in the status bar
- With this PR it should be highlighted and show in the status bar

# Implementation

To do this, I had to make `TeamExplorerContext` aware of the PR mark in the git config, and manually update `IVSGitExt` to notify it of the change. The required making `TextExplorerContext` dependent on `IPullRequestService` in order to not duplicate the code that adds the property to the git config.

### CODE REVIEW

**Comment by jcansdale** [comment]
I've just tried this when creating #1695, but alas it didn't work. 😢 

image

I was on a fork of `VisualStudio` created with the new fork functionality if that might make a difference?

File: src/GitHub.App/Services/PullRequestService.cs [hunk 1]
```diff
@@ -41,14 +41,21 @@ public class PullRequestService : IPullRequestService
...
         readonly IOperatingSystem os;
         readonly IUsageTracker usageTracker;
 
         [ImportingConstructor]
-        public PullRequestService(IGitClient gitClient, IGitService gitService, IOperatingSystem os, IUsageTracker usageTracker)
+        public PullRequestService(
+            IGitClient gitClient,
+            IGitService gitService,
+            IVSGitExt gitExt,
+            IOperatingSystem os,
```
Comment (sguthals) [inline src/GitHub.App/Services/PullRequestService.cs]: Testing

**Comment by StanleyGoldman** [comment]
My first test of this worked

**Comment by StanleyGoldman** [comment]
I also tested this in a fork scenario like @jcansdale mentioned, (A fork I created using the fork function), it worked for me.

**Comment by meaghanlewis** [comment]
This works well for me with regular and forked repositories as well.

**Comment by meaghanlewis** [comment]
Fixes #1686

**Pull request merged**